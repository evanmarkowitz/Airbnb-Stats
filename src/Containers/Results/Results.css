.map-control {
  width: 400px;;
  overflow: hidden;
}