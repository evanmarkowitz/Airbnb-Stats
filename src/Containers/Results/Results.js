import React from 'react'
import { connect } from 'react-redux';
import { Map, GoogleApiWrapper , Marker, InfoWindow} from 'google-maps-react';


import './Results.css'

export const Result = ({ apts, hood, google}) => {
  
  const filterApartments = () => {
    let aptsByHood = apts
    if (hood !== ''){
      aptsByHood = apts.filter(apt => apt.hood === hood)
    }
    return aptsByHood
  }

  const buildApts = () => {
    const aptsToBould = filterApartments()
    return aptsToBould.map((apt,i) => {
      return <Marker key={i} id={i} position={{
        lat: apt.lat,
        lng: apt.long,
      }}
      >
      </Marker>
    })
  }

  const mapStyles = {
    width: '80%',
    height: '80%',
    margin: 'auto',
  };

  return (
    <section className='results'>
      <p>{hood}</p>
      <Map
          google={google}
          zoom={11}
          className='map'
          style={mapStyles}
          initialCenter={{ lat: 40.7527, lng: -73.9943}}
        >
      {buildApts()}
      </Map>
    </section>
  )
}

// const mapStateToProps = (state) => ({
//   apts: state.apts,
//   hood: state.hood
// })

// export  connect(mapStateToProps)(Result)

export default GoogleApiWrapper({
  apiKey: 'AIzaSyBUjpj2w4odf15sphWyngSyGf8VLgPViy0'
})(Result);