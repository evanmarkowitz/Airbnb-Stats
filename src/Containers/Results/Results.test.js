import {Result} from './Results'
import {shallow} from 'enzyme'
import React from 'react'

describe('Result', () => {
  let wrapper;
  let apts
  beforeEach(() => {
  apts = [{reviewsPerMonth: 1.5,
    id: 8252369,
    numReviews: 68,
    price: 85,
    aptType: 'Private room',
    name: 'Cute Sunny Loft Space Williamsburg',
    hood: 'Williamsburg',
    lat: 40.71181346189293,
    long: -73.95663634549271}]
  wrapper = shallow(<Result apts ={apts} hood='Williamsburg' google={1} aptType={'Private room'} setCurrApt ={jest.fn()}/>)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})