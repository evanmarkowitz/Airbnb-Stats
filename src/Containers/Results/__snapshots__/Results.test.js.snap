// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should match the snapshot 1`] = `ShallowWrapper {}`;
